### ISSUE TITLE
RyuJIT/x86: CoreFx System.IO.UnmanagedMemoryStream.TestsExit code = -2146233022

### ISSUE THREAD

**Issue by RussKeldorph** [dotnet/coreclr#8159]
**Repro steps:**

coreclr *(assuming root=c:\git\coreclr)*
```
build x86 checked skiptests
```
corefx *(assuming root=c:\git\corefx)*
```
build -Release -SkipTests -TestNugetRuntimeId=win7-x86 -- /p:BUILDTOOLS_OVERRIDE_RUNTIME=c:\git\coreclr\bin\Product\Windows_NT.x86.Checked
cd bin\tests\AnyOS.AnyCPU.Release\System.IO.UnmanagedMemoryStream.Tests\default.netcoreapp1.1
set COMPlus_AltJit=*
set COMPlus_AltJitName=protojit.dll
RunTests.cmd c:\git\corefx\packages
```
**Output**
```
Using c:\git\corefx\packages as folder for resolving package dependencies.
Executing in C:\git\corefx\bin\tests\AnyOS.AnyCPU.Release\System.IO.UnmanagedMemoryStream.Tests\default.netcoreapp1.1\
Hard linking dependent files...
Finished linking needed files, moving to running tests.
Running tests... Start time: 16:45:09.31
Command(s):
call CoreRun.exe xunit.console.netcore.exe System.IO.UnmanagedMemoryStream.Tests.dll  -xml testResults.xml -notrait Benchmark=true -notrait category=nonnetcoreapp1.1tests  -notrait category=OuterLoop -notrait category=failing -notrait category=nonwindowstests
xUnit.net console test runner (32-bit .NET Core)
Copyright (C) 2014 Outercurve Foundation.

Discovering: System.IO.UnmanagedMemoryStream.Tests
Discovered:  System.IO.UnmanagedMemoryStream.Tests
Starting:    System.IO.UnmanagedMemoryStream.Tests
Finished running tests.  End time=16:45:16.85, Exit code = -2146233022
```

**Comment by mikedn** [comment]
This doesn't appear to be JIT related. `System.Private.CoreLib` is missing a resource string and asserts, `-2146233022` is `COR_E_CODECONTRACTFAILED`. I'll send a PR tomorrow to add the missing string.

**Comment by RussKeldorph** [comment]
@mikedn Thanks.  I admit I didn't compare exactly the same commits between JIT32 and RyuJIT, so it's possible this is a recent regression (due to corelib refactoring?).  Presumably #8158 and #8160 are dups?

@dotnet/jit-contrib

**Comment by mikedn** [comment]
The other 2 appear to be unrelated. It's odd, they show asserts but only when run with a debugger attached.

Not sure what's going on this the missing resource string from this one. Supposedly this was fixed in #7673 but it looks to me that the string was added to the wrong file - mscorlib.txt instead of System.Private.CoreLib.txt.

**Comment by RussKeldorph** [comment]
@alexperovich Any insights?
